File: Data Structures & Algorithms/binary-search/submission-0.go
func search(nums []int, target int) int {
    left, right := 0, len(nums) - 1

    for left <= right {
        middle := (left + right) / 2
        
        if (nums[middle] > target) {
            right = middle - 1
        } else if (nums[middle] < target) {
            left = middle + 1
        } else {
            return middle
        }
    }

    return -1
}

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.js
class Solution {
    /**
     * @param {number} target
     * @param {number[]} nums
     * @return {number}
     */
    minSubArrayLen(target, nums) {
        let currentSum = 0;
        let currentLen = Infinity;
        let left = 0;

        for (let right=0; right<nums.length; right++) {
            currentSum += nums[right];
            while (currentSum >= target) {
                currentLen = Math.min(currentLen, right - left + 1);
                currentSum -= nums[left];
                left++;
            }
        }

        if (currentLen === Infinity) {
            return 0;
        } else {
            return currentLen;
        }
    }
}
